packages/babel-generator/src/node/parentheses.js

```diff
@@ -50,7 +50,7 @@ export function ObjectExpression(node: Object, parent: Object, printStack: Array
     return true;
   }
 
-  return isFirstInStatement(printStack);
+  return isFirstInStatement(printStack, true);
```

Comment:
Has to because we still need the logic that checks it's linear firstness not the structural firstness:

'{}.a' comes first in text (linearly) but structurally MemberExpression is first